drivers/gpio/Kconfig.max22190

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2010-2024 Analog Devices Inc.
+# Copyright (c) 2024 BayLibre SAS
+# SPDX-License-Identifier: Apache-2.0
+
+# MAX22190 GPIO configuration options
+
+menuconfig GPIO_MAX22190
+	bool "MAX22190 GPIO driver"
+	default y
+	depends on DT_HAS_ADI_MAX22190_GPIO_ENABLED
```

Comment:
Need to `select SPI`

Reply:
`select SPI` in Kconfig.max22190 is causing:
 `scripts/kconfig/kconfig.py: Dependency loop`
So I noticed gpio drivers utilizing SPI are using `depends on SPI` instead.